engine/src/main/java/org/terasology/engine/core/module/ModuleManager.java

```diff
@@ -105,49 +98,61 @@ public ModuleManager(String masterServerAddress, List<Class<?>> classesOnClasspa
 
     /**
      * I wondered why this is important, and found MovingBlocks/Terasology#1450.
-     * It's not a worry that the engine module wouldn't be loaded without it. 
-     * It's about ordering: some things run in an order derived from the dependency 
+     * It's not a worry that the engine module wouldn't be loaded without it.
+     * It's about ordering: some things run in an order derived from the dependency
      * tree, and we want to make sure engine is at the root of it.
-     */   
-    private void ensureModulesDependOnEngine(Module engineModule) {
+     */
+    private void ensureModulesDependOnEngine() {
         DependencyInfo engineDep = new DependencyInfo();
         engineDep.setId(engineModule.getId());
         engineDep.setMinVersion(engineModule.getVersion());
-        engineDep.setMaxVersion(engineModule.getVersion().getNextPatchVersion());
+        engineDep.setMaxVersion(engineModule.getVersion().getNextMinorVersion());
```

Comment:
Is this a nother semver related change?

Reply:
oh, it was a workaround to avoid that gestalt#100 problem.

I think it would be okay to not explicitly set a maxVersion here at all, and let the gestalt DependencyInfo make its decisions about what is compatible with the required minVersion.

(But didn't you want to require modules to start specifying their engine compatibility in their module.txt?)

Reply:
I see, that makes sense now. 

And yes, I'd like to have modules specify their engine version (range) explicitly as there is currently no way to figure out with which version of the engine a module can be build and run. Not sure where on the timeline that is, though.

Reply:
Wouldn't be hard to do here on the engine side. It'd be a matter of replacing this "add engine dependencies to everything" method with a "assert things declared their own engine dependency" method.